src/julia/Integer.jl

```diff
@@ -511,9 +511,11 @@ end
 @doc raw"""
     ppio(a::T, b::T)
 
-Split $a$ into $c*d$ where $c = gcd(a, b^\infty)$.
+Return a pair $(c,d)$ such that $a=c*d$ and $c = gcd(a, b^\infty)$ if $a\neq 0$,
+and $c=b$, $d=0$ if $a=0$.
 """
 function ppio(a::T, b::T) where T <: Integer
+   a == 0 && return (b,0)
```

Comment:
```suggestion
   a == 0 && return (b,T(0))
```
I thought that JET would warn about this?

Reply:
Thanks!

And yeah, maybe JET would have caught this, but I did not run JET on this... In an ideal world we could run JET as part of the CI, but we are still quite far from that (`master` has 126 possible errors reported by JET).


BTW this PR does not really reduce the number, because we are still "missing" `ppio` for `Rational`, but I wasn't sure how to implement that given the (to me) strange definition of `gcd` for `Rational` used by Julia.

Reply:
(one option would be to just not do that, and instead change code which calls `ppio` to only accept `Union{RingElem, Integer}` instead of `RingElement`)

Reply:
ppio should accept euc - but that is not a type. We do this all over the place, the type RingElem is a lie in many cases - and this should not be changed I think. this is julias convention: do not be restrictive. If the neccessary functions are there, fine. if not: error

Reply:
In general I agree with this principle, but if used carelessly it makes JET useless for us, and that would be a pity, as it already uncovered several genuine bugs in AA and GAP.jl ...